Add a topic around voice support

bpasero opened the pull request:
fyi @ntrogh @isidorn

ntrogh commented on docs/editor/voice.md:
```diff
@@ -0,0 +1,46 @@
+
+![Editor Dictation Mode](images/accessibility/editor-dictate.png)
+
+You can press and hold the keybinding for the start command (`kb(workbench.action.editorDictation.start)`) to enable walky-talky mode, where the voice recognition stops as soon as you release the keys.
```
```suggestion
You can press and hold the keybinding for the voice start command (`kb(workbench.action.editorDictation.start)`) to enable **walky-talky mode**. Voice recognition is active until you release the keys, after which the request is submitted automatically.
```

ntrogh commented on docs/editor/voice.md:
```diff
@@ -0,0 +1,46 @@
+
+## Voice in GitHub Copilot Chat
+
+You can use your voice to talk to [GitHub Copilot Chat](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-chat) easily. The command **Voice: Start Voice Chat** (`kb(workbench.action.chat.startVoiceChat)`) works everywhere and brings up a voice chat, depending on where your focus is (inline chat in the editor, panel chat otherwise).
```
```suggestion
You can use your voice to talk to [GitHub Copilot Chat](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-chat) easily. The command **Voice: Start Voice Chat** (`kb(workbench.action.chat.startVoiceChat)`) brings up a voice chat, regardless of where the current focus is. If the focus is in the editor, inline chat is started, otherwise the Chat view is opened.
```

ntrogh commented on docs/editor/voice.md:
```diff
@@ -0,0 +1,46 @@
+
+![Voice Chat](images/accessibility/voice-chat.png)
+
+To enable walky-talky mode, press and hold the keybinding. Voice recognition is active until you release the keys, after which the request is submitted automatically.
```
This is a repeat of above. Maybe we can dedicate a section to walky-talky mode?

ntrogh commented on docs/editor/voice.md:
```diff
@@ -0,0 +1,46 @@
+
+To enable walky-talky mode, press and hold the keybinding. Voice recognition is active until you release the keys, after which the request is submitted automatically.
+
+## Using Other Languages
```
```suggestion
## Support for multiple languages
```

ntrogh commented on docs/editor/voice.md:
```diff
@@ -0,0 +1,46 @@
+
+Read on to find out about:
+
+* [Visual Studio Code User Interface](/docs/getstarted/userinterface.md) - A quick orientation to VS Code.
```
```suggestion
* [Other VS Code accessibility features](/docs/editor/accessibility.md).
* [Visual Studio Code User Interface](/docs/getstarted/userinterface.md) - A quick orientation to VS Code.
```

bpasero commented:
@ntrogh thanks, I added your suggestions and added:
* a dedicated section about "Hey Code" (I had forgotten about this!)
* a section about walky talky mode
* explicit mention that speech recognition does not require to be online

isidorn commented on docs/editor/voice.md:
```diff
@@ -0,0 +1,59 @@
+---
+# Voice Support
+
+Microsoft publishes the [VS Code Speech](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-speech) extension to enable various features through the power of only using your voice. Once installed, the extension enables you to dictate into the editor, or to verbally interact with [GitHub Copilot Chat](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-chat).
```
"Through the power of only using your voice" - can we tighten this up?
For example "Through the power of your voice"
I mean you can use this feature in combination with the keyboard/mouse.

isidorn replied:
I find this start weird "Microsoft publishes" as if some giant corp is in charge of this extension. You are the owner :)
Can we just start with "VS Code Speech...

isidorn commented on docs/editor/voice.md:
```diff
@@ -0,0 +1,59 @@
+
+## Voice in GitHub Copilot Chat
+
+You can use your voice to talk to [GitHub Copilot Chat](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-chat) easily. The command **Voice: Start Voice Chat** (`kb(workbench.action.chat.startVoiceChat)`) brings up a voice chat, regardless of where the current focus is. If the focus is in the editor, inline chat is started, otherwise the Chat view is opened.
```
Should we make a point in this section, that Voice works particularly well with GitHub Copilot. Because the language models that are powering GitHub Copilot do not expect 100% precise input?

bpasero replied:
I am not sure its worth to shout this out. This might indicate that we are not confident with the transcription results?

isidorn replied:
That's fair. Let's mark this as resolved.

isidorn commented on docs/editor/voice.md:
```diff
@@ -0,0 +1,59 @@
+
+Each language for the speech extension comes as its own extension. When you start speech recognition for the first time, you will see an extension installation for each language you selected.
+
+## Next steps
```
Should we have a call to action to ask for feedback?
I feel like this feature is still pretty new and we could mention that any feedback is welcome?

Maybe not needed. Not sure.

bpasero replied:
Yeah, though thats probably true for a lot of docs pages. Should there be a generic button in the docs to report issues or feedback? I think that would be cool.

ntrogh replied:
We have feedback buttons in the right sidebar to either ask for features or report issues :)
![image](https://github.com/microsoft/vscode-docs/assets/1908215/5ae84d55-b210-463e-83b5-05e7038ae050)

bpasero replied:
But is that to report issues about the docs or the product?

ntrogh replied:
Those are for product issues and feedback. Docs feedback is with the control at the bottom of the docs page (Yes/No and follow-up verbatim feedback).

isidorn reviewed (approved):
Left minor comments.
Looks great overall!

isidorn commented:
Good to go. Merging in.
